testing/sdk_tasks.py

```diff
@@ -41,26 +41,46 @@ def get_task_ids(service_name, task_prefix):
 
 
 def check_tasks_updated(service_name, prefix, old_task_ids, timeout_seconds=DEFAULT_TIMEOUT_SECONDS):
+    # TODO: strongly consider merging the use of checking that tasks have been replaced (this method)
+    # and checking that the deploy/upgrade/repair plan has completed. Each serves a part in the higher
+    # atomic test, that the plan completed properly where properly includes that no old tasks remain.
     def fn():
         try:
             task_ids = get_task_ids(service_name, prefix)
         except dcos.errors.DCOSHTTPException:
             log.info('Failed to get task ids for service {}'.format(service_name))
             task_ids = []
 
-        log.info('Waiting for tasks starting with "{}" to be updated:\n- Old tasks: {}\n- Current tasks: {}'.format(
-            prefix, sorted(old_task_ids), sorted(task_ids)))
-        all_updated = True
-        for id in task_ids:
-            if id in old_task_ids:
-                all_updated = False
-        if len(task_ids) < len(old_task_ids):
-            all_updated = False
-        return all_updated
+        prefix_clause = ''
+        if prefix:
+            prefix_clause = ' starting with "{}"'.format(prefix)
+
+        old_set = set(old_task_ids)
+        new_set = set(task_ids)
+        newly_launched_set = new_set.difference(old_set)
+        old_remaining_set = old_set.intersection(new_set)
+        # the constrainst of old and new task cardinality match should be covered by completion of
+        # deploy/recovery/whatever plan, not task cardinality, but some uses of this method are not
+        # using the plan, so not the definitive source, so will fail when the finished state of a
+        # plan yields more or less tasks per pod.9
+        all_updated = len(newly_launched_set) == len(new_set) and len(old_remaining_set) == 0 and len(new_set) >= len(old_set)
```

Comment:
Any edge case that I'm failing to see where `len(new_set) > len(old_set)` would be true? If the size of the sets do in fact differ shouldn't that be cause for failure since that's not what we expect?

Reply:
See #1619 for fuller context of this wonkiness due to the overly large set of overloaded uses of this method.

The above was originally a strict definition in the set sense, but the consensus on review of #1619 is/was that the check here is now a good semantic check based on the SDK's meaning for all tasks updated.

I strongly recommend making this method private to sdk_plan as the "all" really means all of the tasks that should have been updated as the result of an action where the action is the execution of a plan. Encapsulating this properly has the positive effect of ensuring that the plan is checked for completion, which includes passing the readiness check, not only that the tasks have cycled.

Reply:
Looks like this change is already in master, so perhaps a rebase is in order before we merge?

Reply:
Oh never mind, I see.